editor/plugins/script_text_editor.cpp

```diff
@@ -790,6 +790,9 @@ void ScriptEditor::_update_modified_scripts_for_external_editor(Ref<Script> p_fo
 		uint64_t date = FileAccess::get_modified_time(script->get_path());
 
 		if (last_date != date) {
+			if (ResourceCache::has(script->get_path())) {
+				ResourceCache::forget(script->get_path());
+			}
```

Comment:
I don't know if that's the right approach, `ResourceLoader::load()` is already called with `p_no_cache = true`, so it's probably a bug that it triggers a cache conflict in this case.

Reply:
Hmm, I hadn't noticed that variable. I've moved the fix to ResourceLoader::load, predicated on p_no_cache. It seems to work just fine (though I'll have to wait for automatic tests go to through), I've only ever managed to trigger ::forget for cs scripts. 

The best way to fix it would probably involve passing p_no_cache to ResourceFormatLoaderCSharpScript::load, but that would require a lot of changes for a relatively small fix. 4.0 effectively does that with CacheMode, but that'd probably require porting back the entire cache system.